simple/app.go

```diff
@@ -88,19 +132,37 @@ type AppManagedKind struct {
 	Watcher operator.ResourceWatcher
 	// Validator is an optional ValidatingAdmissionController for the Kind. It will be run only for validation
 	// of this specific version.
-	Validator resource.ValidatingAdmissionController
+	Validator KindValidator
 	// Mutator is an optional MutatingAdmissionController for the Kind. It will be run only for mutation
 	// of this specific version.
-	Mutator resource.MutatingAdmissionController
-	// SubresourceRoutes are an optional map of subresource paths to a route handler.
+	Mutator KindMutator
+	// CustomRoutes are an optional map of subresource paths to a route handler.
 	// If supported by the runner, calls to these subresources on this particular version will call this handler.
-	SubresourceRoutes map[string]func(ctx context.Context, writer http.ResponseWriter, req *app.SubresourceRequest) error
+	CustomRoutes []AppCustomRouteHandler
 	// ReconcileOptions are the options to use for running the Reconciler or Watcher for the Kind, if one exists.
-	ReconcileOptions ReconcileOptions
+	ReconcileOptions BasicReconcileOptions
 }
 
-// ReconcileOptions are settings for the ListWatch and informer setup for a reconciliation loop
-type ReconcileOptions struct {
+type AppCustomRouteHandler struct {
+	Path    string
+	Methods []string
```

Comment:
That works, I was just using similar syntax to a lot of popular routers that let you attach multiple methods to a handler. We could always change this in the future if we need to as well, as this is the part that can evolve since there are no current runners that support custom routes.

Reply:
> That works, I was just using similar syntax to a lot of popular routers that let you attach multiple methods to a handler.

Yeah, I get where you were coming from, I just personally think that it usually is not what devs want. Plus, it's always easier to relax the requirements down the line than to increase them.